### DISCUSSION TITLE
What is the intended way of saving the base model when finetuning both an adapter and some layers in the base model?

### DISCUSSION THREAD

**Discussion by samedii** [huggingface/peft#1546] [Q&A] [answered]
Is there are better way? I feel like I might be missing some function in the API.

```
unet.save_pretrained("lora")
unet.unload().save_pretrained("unet")
unet = peft.PeftModel.from_pretrained(
    unet, "lora"
)
```

This is problematic because it breaks the optimizer each time I save the model.

**Comment by samedii** [comment]
Ended up doing this but hoping there will be a more native solution in the future.

```
            model.denoiser.save_pretrained("output/denoiser-peft")

            torch.save(
                {
                    name: parameter
                    for name, parameter in model.denoiser.named_parameters()
                    if parameter.requires_grad and "lora" not in name
                },
                "output/denoiser-trainable.pt",
            )
```

**Reply by younesbelkada** [reply, marked as answer]
Hi @samedii 
in order to save both the adapter weights and some layers of the base model, that I assume are trainable, you need to declare a `modules_to_save` variable in your PeftConfig and peft will automatically take care of saving / loading correct modules.
